ca_gazebo/src/cliff_msg_publisher.cpp

```diff
@@ -0,0 +1,118 @@
+  GZ_REGISTER_MODEL_PLUGIN(GazeboCliffMsgPublisher)
+
+  void GazeboCliffMsgPublisher::Load(physics::ModelPtr _parent, sdf::ElementPtr _sdf)
+  {
+    this->update_period_ = 1.0/(_sdf->Get<double>("updateRate"));
+    // Make sure the ROS node for Gazebo has already been initialized
+    if (!ros::isInitialized())
+    {
+      ROS_FATAL_STREAM("A ROS node for Gazebo has not been initialized, unable to load plugin. "
+        << "Load the Gazebo system plugin 'libgazebo_ros_api_plugin.so' in the gazebo_ros package)");
+      return;
+    }
+    if (!ros::isInitialized()) 
+    {
+      int argc = 0;
+      char** argv = NULL;
+      ros::init(argc, argv, "cliff_msg_publisher");
+    }
+    // Initialize the ros variables and gazebo variables
+    this->robot_number_ =_sdf->Get<char>("robotNumber");
+    if (!_sdf->HasElement("frameName"))
+    {
+        ROS_INFO("cliff plugin missing <frameName>, defaults to world");
+        this->frame_name_ = "world";
+    }
+    else
+        this->frame_name_ = _sdf->Get<std::string>("frameName");
+
+    std::ostringstream oss;
+    oss << "create" << this->robot_number_ << "/side_left_cliff_sensor/raw/";
+    std::string topic_name = oss.str();
+    this->side_left_ = this->nh_.subscribe(topic_name, 1, &GazeboCliffMsgPublisher::SideLeftCliffCallback, this);
+    oss.str("");
+    oss << "create" << this->robot_number_ << "/side_right_cliff_sensor/raw/";
+    topic_name = oss.str();
+    this->side_right_ = this->nh_.subscribe(topic_name, 1, &GazeboCliffMsgPublisher::SideRightCliffCallback, this);
+    oss.str("");
+    oss << "create" << this->robot_number_ << "/front_left_cliff_sensor/raw/";
+    topic_name = oss.str();
+    this->front_left_ = this->nh_.subscribe(topic_name, 1, &GazeboCliffMsgPublisher::FrontLeftCliffCallback, this);
+    oss.str("");
+    oss << "create" << this->robot_number_ << "/front_right_cliff_sensor/raw/";
+    topic_name = oss.str();
+    this->front_right_ = this->nh_.subscribe(topic_name, 1, &GazeboCliffMsgPublisher::FrontRightCliffCallback, this);
+    oss.str("");                  
+    oss << "create" << this->robot_number_ << "/cliff_msg/";
+    topic_name = oss.str();
+    this->publisher_ = this->nh_.advertise<ca_msgs::Cliff>(topic_name, 1);
+    this->updateConnection_ = event::Events::ConnectWorldUpdateBegin(
+      std::bind(&GazeboCliffMsgPublisher::OnUpdate, this));    
+    this->prev_update_time_ = ros::Time::now();
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////
+  // Update the controller
+  void GazeboCliffMsgPublisher::OnUpdate()
+  //Every update check if one of the sensors is active, if it is, change the output to true, publishes every 1ms
+  {
+    if ((ros::Time::now() - this->prev_update_time_) < ros::Duration(this->update_period_)) 
+    {
+      return;
+    }
+    ca_msgs::Cliff msg;
+    msg.header.frame_id = this->frame_name_;
+    msg.header.seq = this->seq_counter_;
```

Comment:
We talked about this during our last meeting:

1. `ca_msgs::Cliff msg;`: declare this as a class member.
2. `msg.header.frame_id = this->frame_name_;`: can be declared inside the Load.
3. `msg.header.seq = this->seq_counter_;`: `this->msg.header.seq++`

Reply:
Maybe I didn't catch it because I asked if I should put `ca_msgs::Cliff msg;` as a member of the class and I thought you said no, and since I was creating the message every time, I couldn't put `msg.header.frame_id = this->frame_name_` on the load or do the adding of the `msg.header.seq ` directly, I remember asking you about that in specific. 

But I'll do it the way that is commented here.